addons/sale/tests/test_sale_order_down_payment.py

```diff
@@ -1073,3 +1073,46 @@ def test_downpayment_description(self):
         reversal_move.action_post()
         self.assertEqual(reversal_move.move_type, 'out_refund')
         self.assertIn('ref', so_dp_line.name)
+
+    def test_downpayment_excludes_confirmed_and_draft_invoices(self):
+        self.sale_order.order_line[0].product_id = self.company_data['product_service_order'].id
+        self.sale_order.order_line[0].product_uom_qty = 1
+        self.sale_order.order_line[0].price_unit = 100
+
+        self.sale_order.order_line[1].product_id = self.company_data['product_delivery_no'].id
+        self.sale_order.order_line[1].product_uom_qty = 1
+        self.sale_order.order_line[1].qty_delivered = 0
+        self.sale_order.order_line[1].tax_id = self.tax_15
+        self.sale_order.order_line[1].price_unit = 100
+
+        self.sale_order.order_line[2:].unlink()
+
+        self.sale_order.action_confirm()
+        # One draft invoice is created for the pre_paid product.
+        self.sale_order._create_invoices()
+
+        so_context = {
+            'active_model': 'sale.order',
+            'active_ids': [self.sale_order.id],
+            'active_id': self.sale_order.id,
+            'default_journal_id': self.company_data['default_journal_sale'].id,
+        }
+        payment_params = {
+            'advance_payment_method': 'fixed',
+            'fixed_amount': 50.0,
+            'deposit_account_id': self.revenue_account.id,
+        }
+        downpayment = self.env['sale.advance.payment.inv'].with_context(so_context).create(payment_params)
```

Comment:
```suggestion
        downpayment = self.env['sale.advance.payment.inv'].with_context(**so_context).create({
            'advance_payment_method': 'fixed',
            'fixed_amount': 50.0,
            'deposit_account_id': self.revenue_account.id,
        })
```
Line too long, and let's add to the context instead of replacing it.